packages/devtools_app/lib/src/screens/performance/panes/queued_microtasks/queued_microtasks_controller.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright 2025 The Flutter Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file or at https://developers.google.com/open-source/licenses/bsd.
+
+import 'dart:async';
+
+import 'package:devtools_app_shared/utils.dart';
+import 'package:flutter/foundation.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../../../shared/globals.dart';
+import '../../../../shared/utils/future_work_tracker.dart';
+import '../../performance_controller.dart';
+import '../../performance_model.dart';
+import '../flutter_frames/flutter_frame_model.dart';
+
+enum QueuedMicrotasksControllerStatus { empty, refreshing, ready }
+
+class QueuedMicrotasksController extends PerformanceFeatureController
+    with AutoDisposeControllerMixin {
+  QueuedMicrotasksController(super.controller) {
+    addAutoDisposeListener(_refreshWorkTracker.active, () {
+      final active = _refreshWorkTracker.active.value;
+      if (active) {
+        _status.value = QueuedMicrotasksControllerStatus.refreshing;
+      } else {
+        _status.value = QueuedMicrotasksControllerStatus.ready;
+      }
+    });
+  }
+
+  final _status = ValueNotifier<QueuedMicrotasksControllerStatus>(
+    QueuedMicrotasksControllerStatus.empty,
+  );
+  ValueListenable<QueuedMicrotasksControllerStatus> get status => _status;
+
+  final _queuedMicrotasks = ValueNotifier<QueuedMicrotasks?>(null);
+  ValueListenable<QueuedMicrotasks?> get queuedMicrotasks => _queuedMicrotasks;
+
+  final _selectedMicrotask = ValueNotifier<Microtask?>(null);
+  ValueListenable<Microtask?> get selectedMicrotask => _selectedMicrotask;
+
+  final _refreshWorkTracker = FutureWorkTracker();
+
+  @override
+  void onBecomingActive() {}
+
+  Future<void> refresh() => _refreshWorkTracker.track(() async {
+    _selectedMicrotask.value = null;
+
+    final isolateId = serviceConnection
+        .serviceManager
+        .isolateManager
+        .selectedIsolate
+        .value!
+        .id!;
+    final queuedMicrotasks = await serviceConnection.serviceManager.service!
+        .getQueuedMicrotasks(isolateId);
+    _queuedMicrotasks.value = queuedMicrotasks;
+
+    return;
+  });
+
+  void setSelectedMicrotask(Microtask? microtask) {
+    _selectedMicrotask.value = microtask;
+  }
+
+  @override
+  Future<void> handleSelectedFrame(FlutterFrame frame) async {}
+
+  @override
+  Future<void> setOfflineData(OfflinePerformanceData offlineData) async {}
```

Comment:
OfflinePerformanceData should be able to take the data from this tab and serialize it for offline exports of the performance page data. Then this method should set the `_queuedMicrotasks` and `_selectedMicrotask` from offline data when this method is called. See the other performance feature controllers for an example.